react-workshop/src/components/news.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from "react";
+import NewsItem from "./news-item";
+import SearchBar from "./search-bar";
+import Button from "./button";
+
+class NewsList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      search: "",
+      list: props.list
+    };
+  }
+
+  showAlert() {
+    alert("Moggu homo");
+  }
+
+  onChangeSearch = search => this.setState({ search });
+
+  addNews(event) {
+    event.preventDefault();
+    let title = this.refs.title.value;
+    let author = this.refs.author.value;
+
+    let objectID = Math.floor(Math.random() * 100 + 1);
+
+    this.setState({
+      list: this.state.list.concat({
+        objectID,
+        title,
+        author
+      })
+    });
+    this.refs.title.value = "";
```

Comment:
refs niet gebruiken https://stackoverflow.com/questions/40568385/whats-a-bad-practice-with-refs-in-react Refs is een bad practice